asset/asset.go

```diff
@@ -1499,23 +1499,32 @@ func New(genesis Genesis, amount, locktime, relativeLocktime uint64,
 }
 
 // TapCommitmentKey is the key that maps to the root commitment for a specific
-// asset group within a TapCommitment.
+// asset within a TapCommitment.
 //
 // NOTE: This function is also used outside the asset package.
-func TapCommitmentKey(assetID ID, groupKey *btcec.PublicKey) [32]byte {
-	if groupKey == nil {
-		return assetID
+func TapCommitmentKey(assetSpecifier Specifier) [32]byte {
+	var commitmentKey [32]byte
+
+	switch {
+	case assetSpecifier.HasGroupPubKey():
+		assetSpecifier.WhenGroupPubKey(func(pubKey btcec.PublicKey) {
+			serializedPubKey := schnorr.SerializePubKey(&pubKey)
+			commitmentKey = sha256.Sum256(serializedPubKey)
+		})
+
+	case assetSpecifier.HasId():
```

Comment:
Since technically there could be an asset specifier that has neither the ID or the group key (of course only due to a mistake by a dev), should we make this a bit more fool proof and move the `TapCommitmentKey` to be a method on the asset specifier? And then panic if neither of the values are set?

Reply:
I'm not sure. I think we should try to keep `Specifier` and its methods lean. To the same affect, what do you think about adding a default clause to this `switch` which panics?

Reply:
Yeah, sounds good to me :+1: